### DISCUSSION TITLE
Renaming storages_invested etc. to node_invested

### DISCUSSION THREAD

**Discussion by mihlema** [spine-tools/SpineOpt.jl#440] [Ideas]
Currently, we have a bit of a mix in calling `node` `nodes` or `storages`, mostly as we talk about investment decisions. I wonder if we want to unify all of this and simply call it `node` everywhere.

Related to this we also call our variables `connection/unit` and `connections/units` somewhat arbitrarily. I'm not sure if there is a solution (e.g. always using the plural), that we could use to assimilate this.

**Comment by DillonJ** [comment]
Regarding storage investments, it is this way because there is an underlying difference - for connections and units, if not invested in, they are not part of the actual system. But in our structure, storage is a property of a node and the investment decision only applies to the storage of the node and not the node itself which exists regardless of the investment decision. Sometimes we need a handle on the node to which the candidate storage applies and in this case we use node. Perhaps we could use something like storage_node to be more clear?

I don't have a strong view on the plural/singular issue but I think you're right in that we need to be consistent.

**Comment by mihlema** [comment]
Thanks, @DillonJ. Yes, I think we also talked about this at some point, and I see why it is there. But then it is also seems somewhat inconsistent if we use `node_investment_time_indices` imo. But I do see the need to be able to differentiate (e.g. if we want to allow the model to build a new bus etc. or something in the future). 

But I think we could combine both somehow? We could think of the `storage_node` in a way, that technically it also doesn't exist before it is invested in. And vise versa a new location can be a new "storage" node, with 0 capacity. Probably needs some more thinking-it-through, but it would be nice if we could somehow use ~the same code for both cases.

**Reply by DillonJ** [reply]
The way I was thinking when I named stuff was I tried to be true to the high level structure. I.e. storage is a property of a node and the temporal structure follows from the node and not the storage. Just say you made demand investible for example, you would use the same node_investment_time_indices. Unless of course you wanted a different investment temporal structure for demands and storages in this case, and then perhaps you need `storage_investment_time_indices` and `demand_investment_time_indices`. So I suppose you could argue it either way. 

I think if we really want to disassociate storage with nodes then the right way to do it is to have a storage object - but right now, storage is a property of a node - I don't think we can assume what a user wants to do with a node that has investment storages in the current implementation - I can see cases where the node is meaningful without the storage and others where the node is there just to host the storage.

Maybe one day, perhaps when we have artchetypes, we might want to revisit the structure. I think the main rationale originally was that having a distinct storage object was a little cumbursome when it came to doing simple things.

Edit: by the way - really just outlining my thought process here - I don't have strong feelings about this :-)
